onnx/hub.py

```diff
@@ -363,12 +363,25 @@
                 "download the model from the model hub."
             )
 
+    def members_filter(tarfile, base):
+        result = []
+        for member in tarfile:
+            member_path = os.path.join(base, member.name)
+            abs_base = os.path.abspath(base)
+            abs_member = os.path.abspath(member_path)
+            if not abs_member.startswith(abs_base):
+                raise AssertionError("The tarball file in downloading model contains directory traversal sequence which may contain harmful payload.")
+            elif member.issym() or member.islnk():
+                raise AssertionError("The tarball file in downloading model contains symbolic links which may contain harmful payload.")
+            result.append(member)
+        return result
     with tarfile.open(local_model_with_data_path) as model_with_data_zipped:
         # FIXME: Avoid index manipulation with magic numbers
         local_model_with_data_dir_path = local_model_with_data_path[
             0 : len(local_model_with_data_path) - 7
         ]
-        model_with_data_zipped.extractall(local_model_with_data_dir_path)
+        # FIX tarball directory traversal 
+        model_with_data_zipped.extractall(path=local_model_with_data_dir_path, members=members_filter(model_with_data_zipped, local_model_with_data_dir_path))
```

Comment:
Codeql detected the source of the Tarball is from remote sources, but sanitization of Tarball extraction is performed to prevent arbitrary file overwrite.